File: go/demo/co.js
function sleep(ms){
  return new Promise((resolve,reject) => setTimeout(
     ()=>resolve(),ms
  ))
}

async function hello(){
  await sleep(3000)
  console.log("hello")
};

async function test(){
  await sleep(3000)
  console.log("world")
};

await hello();
await test();

File: go/demo/co.go
package main

import (
  "fmt"
  "time"
)

func deferPrint(str string){
  time.Sleep(time.Second*2)
  fmt.Println(str)
}

func main(){
  go deferPrint("hello")
  go deferPrint("world")
  //如果主协程不阻塞，永远不会切换

  time.Sleep(time.Second*4)
}
